### PR TITLE
Nerfs the roundstart planetary gasmixes, cleans up the code a bit

### PR DESCRIPTION
**Pull request by LemonInTheDark**
## About The Pull Request

When floyd added station traits, he made the unusual atmosphere trait uncap the amount of restricted gas that can be generated. I didn't put as much thought into it as I should have, as all restricted gases are in low enough concentrations that they cannot react on their own, this prevents infinite plasmafires and such.

In light of that, I removed it from that check. However, this means the trait doesn't do a whole lot, since restricted chance for both icemoon and lavaland is 50, which means most of the time they'll roll most if not all their restricted gases.

The probability on this is actually a bit hard to calculate, since not only does rolling a restricted gas increase the amount of chances you have to roll a restricted gas, it's in theory possible to roll restricted gasses every other time, and only get one type out of it. I've removed this aspect by making it pick and subtract from an existing list.

In any case, I also lowered the base chance of rolling restricted gases, and increased the trait additive. I hope this will make seeing odd gasses on icebox and lavaland more rare, and make the stationtrait feel more impactful without causing hell behavior

## Changelog
:cl:
tweak: Planetary gasmixes are less likely to be filled with rare gasmixes now, unless the right station trait rolls.
fix: Lavaland and icebox can no longer cause infinite fires in rare edgecases
/:cl:

### CODE REVIEW

**Comment by TemporalOroboros** [comment]
Doesn't this mean that there can only be at most 0.2 kPa of water vapor or other gases since they can't ever roll for more even with the station trait?

File: code/datums/atmosphere/_atmosphere.dm [hunk 1]
```diff
@@ -36,14 +37,13 @@
...
 			gastype = pick(normal_gases)
 			amount = normal_gases[gastype]
 		else
-			gastype = pick(restricted_gases)
-			amount = restricted_gases[gastype]
-			if(!HAS_TRAIT(SSstation, STATION_TRAIT_UNNATURAL_ATMOSPHERE) && gaslist[gastype])
-				continue
+			gastype = pick(spicy_gas)
+			amount = spicy_gas[gastype]
+			spicy_gas -= gastype //You can only pick each resricted gas once
```
Comment (Jared-Fogle) [inline code/datums/atmosphere/_atmosphere.dm:46]:
```suggestion
			spicy_gas -= gastype //You can only pick each restricted gas once
```

**Comment by LemonInTheDark** [comment]
> 
> 
> Doesn't this mean that there can only be at most 0.2 kPa of water vapor or other gases since they can't ever roll for more even with the station trait?

Moles not pressure, but yes

File: code/datums/atmosphere/_atmosphere.dm (hunk 1 above)
Comment (LemonInTheDark) [reply code/datums/atmosphere/_atmosphere.dm:46]: Haha I can totally spell, did you know english is my first language?

**Comment by Qustinnus** [comment]
Why don't you just add a cap to the amount of any gas?

**Comment by Qustinnus** [comment]
optionally, make it so that there's a list of new /datum/atmosphere the trait can pick from instead of the normal one, which have an extremely alternate atmosphere set but won't automatically be super dangerous.

**Comment by LemonInTheDark** [comment]
> 
> 
> Why don't you just add a cap to the amount of any gas?

Because that should be what the restricted list does in the first place, I don't want to make adding new gasmixes any harder then it is currently

> 
> 
> optionally, make it so that there's a list of new /datum/atmosphere the trait can pick from instead of the normal one, which have an extremely alternate atmosphere set but won't automatically be super dangerous.

Do you have any suggestions?

**Comment by Mothblocks** [comment]
i trust mr. atmos

**Pull request merged**